WinRT.Runtime/Marshalers.cs

```diff
@@ -936,6 +936,16 @@ public static IObjectReference CreateMarshaler(T o, bool unwrapObject = true)
             {
                 return objRef.As<IInspectable.Vftbl>();
             }
+            var publicType = o.GetType();
+            Type helperType = Projections.FindCustomHelperTypeMapping(publicType, true);
+            if(helperType != null)
+            {
+                var parms = new[] { Expression.Parameter(typeof(object), "arg") };
+                var createMarshaler = Expression.Lambda<Func<object, IObjectReference>>(

```

Comment:
We may want to open an issue to look at how expensive this is both here and below in the FromAbi case and whether we should consider caching it to improve performance.